src/backend/marsha/core/serializers/video.py

```diff
@@ -286,6 +286,117 @@ class InitLiveStateSerializer(serializers.Serializer):
     type = serializers.ChoiceField(LIVE_TYPE_CHOICES)
 
 
+class LiveRegistrationSerializer(serializers.ModelSerializer):
+    """Serializer for liveRegistration model."""
+
+    class Meta:  # noqa
+        model = LiveRegistration
+        fields = (
+            "email",
+            "id",
+            "consumer_site",
+            "lti_user_id",
+            "should_send_reminders",
+            "video",
+        )
+        read_only_fields = (
+            "id",
+            "consumer_site",
+            "lti_user_id",
+            "video",
+        )
+
+    # Make sure video UUID is converted to a string during serialization
+    video = serializers.PrimaryKeyRelatedField(
+        read_only=True, pk_field=serializers.CharField()
+    )
+
+    def validate(self, attrs):
+        """Control or set data with token informations.
+
+        Force the video field to the video of the JWT Token if any.
+        Check email, if present in the token, is equal to the one in the request.
+        Set lti informations if they are present in the token. Control integrity
+        errors and set specific messages.
+
+        Parameters
+        ----------
+        data : dictionary
+            Dictionary of the deserialized values of each field after validation.
+
+        Returns
+        -------
+        dictionary
+            The "data" dictionary is returned after modification.
+
+        """
+        # User here is a video as it comes from the JWT
+        # It is named "user" by convention in the `rest_framework_simplejwt` dependency we use.
+        user = self.context["request"].user
+        video = Video.objects.get(pk=user.id)
+        if video.is_scheduled is False:
+            raise serializers.ValidationError(
+                {"video": f"video with id {user.id} doesn't accept registration."}
+            )
+
+        if not attrs.get("video_id") and isinstance(user, TokenUser):
+            attrs["video_id"] = user.id
+            # consumer_site is defined if context_id exists in the token
+            attrs["consumer_site"] = (
+                Playlist.objects.get(
+                    lti_id=user.token.payload["context_id"]
+                ).consumer_site
+                if user.token.payload.get("context_id") is not None
+                else None
+            )
+
+            if user.token.payload.get("user") is not None:
+                attrs["lti_user_id"] = user.token.payload["user"]["id"]
+
+                # If email is present in token, we make sure the one sent is the one expected
+                if user.token.payload["user"].get("email") is not None:
+                    if attrs["email"] != user.token.payload["user"].get("email"):
+                        raise serializers.ValidationError(
+                            {"email": f"{attrs['email']} is not the expected email."}
```

Comment:
In fact it seems weird that we ask them to submit the email but in fact we already know what we expect ^^
So should we check that the email field is blank and if not raise an error that is more like:

> you are not authorized to register with a specific email. We will use the email from you authentication